src/Relude/List/Reexport.hs

```diff
@@ -8,18 +14,163 @@ License:    MIT
 Maintainer: Kowainik <xrom.xkov@gmail.com>
 
 Reexports most of the "Data.List" and "Data.List.NonEmpty".
+
+Note, that Relude reexports 'head', 'tail', 'init', 'last' from
+"Data,List.NonEmpty" instead of the "Data.List", so these functions are safe to
+use.
+
+`relude` also provides custom type error for better experience with transition
+from lists to 'NonEmpty' with those functions.
 -}
 
 module Relude.List.Reexport
        ( module Data.List
-       , module Data.List.NonEmpty
-       , module GHC.Exts
+         -- * NonEmpty List
+       , NonEmpty (..)
+       , nonEmpty
+       , head
+       , init
+       , last
+       , tail
+       , sortWith
```

Comment:
Nice! I love more explicit import list here. A few ideas: can we add section `-- * List` in front of `Data.List` and move `sortWith` directly after `module Data.List` reexport?